packages/linter/src/builders/lint/lint.impl.ts

```diff
@@ -1,23 +1,368 @@
-import {
-  BuilderContext,
-  BuilderOutput,
-  createBuilder
-} from '@angular-devkit/architect';
+import { createBuilder } from '@angular-devkit/architect';
+import eslint, { CLIEngine } from 'eslint';
+import { existsSync, readFileSync, writeFileSync } from 'fs';
+import * as glob from 'glob';
+import { Minimatch } from 'minimatch';
+import * as path from 'path';
+import * as ts from 'typescript';
+import { BuilderContext, BuilderOutput } from '@angular-devkit/architect';
 import { from, Observable } from 'rxjs';
-import { concatMap } from 'rxjs/operators';
+
+export function stripBom(data: string) {
+  return data.replace(/^\uFEFF/, '');
+}
+
+async function _loadESLint() {
+  let eslint;
+  try {
+    eslint = await import('eslint');
+    return eslint;
+  } catch {
+    throw new Error('Unable to find ESLint. Ensure ESLint is installed.');
+  }
+}
+
+/**
+ * - Copied from TSLint source:
+ *
+ * Returns an array of all outputs that are not `undefined`
+ */
+export function mapDefined<T, U>(
```

Comment:
Is it worth splitting this file up so that there's one function per file? We can then have a specfile per function?

Reply:
agree :+1: that would make things more organised/easier to test